Ecs/src/SparseArray.hpp

```diff
@@ -0,0 +1,268 @@
+/*
+** EPITECH PROJECT, 2024
+** R-Type Server
+** File description:
+** SparseArray
+*/
+
+#pragma once
+
+#include <optional>
+#include <vector>
+#include <memory>
+
+template <typename Component>
+class SparseArray {
+
+    public :
+        using value_type = std::optional<Component>;
+        using reference_type = value_type &;
+        using const_reference_type = value_type const &;
+        using container_t = std::vector<value_type>;
+        using size_type = typename container_t::size_type ;
+        using iterator = typename container_t::iterator ;
+        using const_iterator = typename container_t::const_iterator ;
+
+    public :
+
+        /**
+         * @brief Construct a new SparseArray object.
+         *
+         */
+        SparseArray (){};
+
+        /**
+         * @brief Construct a new SparseArray object.
+         * It's a copy constructor.
+         *
+         * @param other SparseArray to copy.
+         */
+        SparseArray (SparseArray const &other)
+        {
+            this->_data = other._data;
```

Comment:
I think it would be a great idea to discuss the usage of "this" to have the same logic in all the project. For me, it's useless to use the "this" specifier for private variables, since their type is already defined by the underscore before the variable name. I'll discuss with Axel about this today 👍

Reply:
I'll not say this for the all this specifiers, but you know what i mean

Reply:
Finally, I only use this-> for method call and not private variables since their type is already defined by the underscore.

Reply:
Seen with @AxelF44, I'll update the Contributing soon